scripts/taxonomy/metadata_models.py

```diff
@@ -0,0 +1,318 @@
+    _OPENCONTEXT_SAMPLE_MODEL = None
+
+    @staticmethod
+    def load_model_from_path(collection, label_type):
+        """
+            Set the pretrained models by loading them from the file system
+            Prerequisite: In order to use this, make sure that there is a pydantic settings file on the
+            at the root of this repository named "isamples_web_config.env" with the config file path, and the model path
+            defined
+
+            :param collection : the collection type of the sample
+            :param label_type : the field that we want to predict
+        """
+        # load the model config file
+        if collection == "SESAR" and label_type == "material":
+            config_path = config.Settings().sesar_material_config_path
+        elif collection == "OPENCONTEXT" and label_type == "material":
+            config_path = config.Settings().opencontext_material_config_path
+        elif collection == "OPENCONTEXT" and label_type == "sample":
+            config_path = config.Settings().opencontext_sample_config_path
+        # read the model config file
+        if not os.path.exists(config_path):
+            logging.error(
+                "Unable to locate pretrained models",
+            )
+            return
+        with open(config_path) as json_file:
+            config_json = json.load(json_file)
+        # use the model config to get the pretrained model
+        model = Model(config_json)
+
+        # initialize the model fields
+        if collection == "SESAR" and label_type == "material":
+            MetadataModelLoader._SESAR_MATERIAL_MODEL = model
+        elif collection == "OPENCONTEXT" and label_type == "material":
+            MetadataModelLoader._OPENCONTEXT_MATERIAL_MODEL = model
+        elif collection == "OPENCONTEXT" and label_type == "sample":
+            MetadataModelLoader._OPENCONTEXT_SAMPLE_MODEL = model
+
+    @staticmethod
+    def initialize_models():
+        """
+            Invokes the load_model function to load all of the possible models
+            that are available based on the config
+        """
+        MetadataModelLoader.load_model_from_path("SESAR", "material")
+        MetadataModelLoader.load_model_from_path("OPENCONTEXT", "material")
+        MetadataModelLoader.load_model_from_path("OPENCONTEXT", "sample")
+
+    @staticmethod
+    def get_sesar_material_model():
+        return MetadataModelLoader._SESAR_MATERIAL_MODEL
+
+    @staticmethod
+    def get_oc_material_model():
+        return MetadataModelLoader._OPENCONTEXT_MATERIAL_MODEL
+
+    @staticmethod
+    def get_oc_sample_model():
+        return MetadataModelLoader._OPENCONTEXT_SAMPLE_MODEL
+
+
+class SESARMaterialPredictor:
+    """Material label predictor of SESAR collection"""
+    def __init__(self, model: Model):
+        self._model = model
+        self._model_valid = model is not None
+        self._description_map = None
+
+    def checkInformative(self, text):
```

Comment:
Please convert method name to `check_informative` per https://peps.python.org/pep-0008/#function-and-variable-names

I realize that not all of our code currently follows this but I've been trying to enforce it on newly added code.

Please add typing information to the text method parameter.

Once this is in `isamples_metadata`, I suspect `mypy` will start complaining about the lack of typing information.  I explicitly added type checking in https://github.com/isamplesorg/isamples_inabox/pull/218